app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,35 @@ def show
     @post = Post.find(params[:id])
     @user = @post.author
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = current_user.posts.create(post_params)
+
+    if @post.valid?
+      redirect_to user_post_path(current_user, @post), notice: t('.created')
+    else
+      redirect_to new_user_post_path(current_user), alert: t('.not_created')
+    end
+  end
+
+  def add_like
+    @post = Post.find(params[:id])
+    Like.create(post: @post, author: current_user)
+    Like.update_like_counter(@post)
+    redirect_to user_posts_path(@post.author)
+  end
+
```

Comment:
- [ ] This like implementation should be in its own Like controller, not the Post controller. Kindly create a like controller and move this to that file under the `create` action.